src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -1154,12 +1163,19 @@ public void assertGoodStatus(Page page) {
     }
 
     public <R extends Run> R assertBuildStatusSuccess(Future<? extends R> r) throws Exception {
-        assertThat("build was actually scheduled", r, Matchers.notNullValue());
-        return assertBuildStatusSuccess(r.get());
+        return assertBuildStatus(Result.SUCCESS, r);
     }
 
-    public <J extends AbstractProject<J,R>,R extends AbstractBuild<J,R>> R buildAndAssertSuccess(J job) throws Exception {
-        return assertBuildStatusSuccess(job.scheduleBuild2(0));
+    public <J extends Job<J,R>,R extends Run<J,R>> R buildAndAssertSuccess(final J job) throws Exception {
+        assertThat(job, IsInstanceOf.instanceOf(ParameterizedJobMixIn.ParameterizedJob.class));
+        QueueTaskFuture f = new ParameterizedJobMixIn() {
```

Comment:
Note that there is a new `static` convenience method in 1.621, but

* our baseline is 1.580.x
* I could not get it to compile anyway—generics hell